specification/machinelearningservices/data-plane/Microsoft.MachineLearningServices/preview/2019-05-01/execution.json

```diff
@@ -0,0 +1,1103 @@
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "runDetails",
+            "in": "body",
+            "description": "The details of the run to cancel.",
+            "required": false,
+            "schema": {
+              "$ref": "#/definitions/ExperimentRunDetails"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The run was successfully cancelled.",
+            "schema": {
+              "$ref": "#/definitions/RunStatus"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Cancel a run": {
+            "$ref": "./examples/cancelRun.json"
+          }
+        }
+      }
+    },
+    "/execution/v1.0/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearningServices/workspaces/{workspaceName}/experiments/{experimentName}/runId/{runId}/cancel": {
+      "post": {
+        "tags": [
+          "Execution"
+        ],
+        "summary": "Cancel a run.",
+        "description": "Cancels a run within an experiment.",
+        "operationId": "Execution_CancelRunWithUri",
+        "consumes": [],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/workspaceNameParameter"
+          },
+          {
+            "name": "experimentName",
+            "in": "path",
+            "description": "The experiment name.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "runId",
+            "in": "path",
+            "description": "The id of the run to cancel.",
+            "required": true,
+            "type": "string",
+            "pattern": "^[a-zA-Z0-9][\\w-]{0,255}$"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The run was successfully cancelled.",
+            "schema": {
+              "$ref": "#/definitions/RunStatus"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Cancel a run with URI": {
+            "$ref": "./examples/cancelRunWithUri.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "RunStatus": {
+      "description": "Contains the details of a run.",
+      "type": "object",
+      "properties": {
```

Comment:
Are neither `runId` nor `target` required properties? As defined, `{}` is a valid `RunStatus` model...

Reply:
@johanste Runstatus definition is not used as request payload any where in the service and is only used as response payload for APIs. Do we need to mark properties as Required and read-only for all definitions or just the once which are used as request payload

Reply:
We can do some easy improvements here:
1) Rename RunStatus to StartRunResult or something more meaningful. It's not a status anymore. :)
2) Remove Target from either the structure or the documentation

Reply:
@wbaumann-microsoft 
1. I will rename it.
2. Do you mean remove the Target property from RunStatus(StartRunResult) ?

Reply:
Yes, I do -- fairly certain we have no client behavior that depends on that field, it's a legacy from before public preview.

Reply:
Even if the API still returns it, we shouldn't document it, because we do want to get rid of it. :)

Reply:
done